src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -1391,15 +1403,15 @@ internal bool CanSaveAs(object parameters)
         /// <param name="path">The path to the file.</param>
         /// <exception cref="IOException"></exception>
         /// <exception cref="UnauthorizedAccessException"></exception>
-        internal void SaveAs(string path)
+        internal void SaveAs(string path, bool isBackup = false)
         {
             try
             {
                 Model.Logger.Log(String.Format(Properties.Resources.SavingInProgress, path));
 
-                CurrentSpaceViewModel.Save(path, Model.EngineController);
-                
-                AddToRecentFiles(path);
+                CurrentSpaceViewModel.Save(path, isBackup, Model.EngineController);
+
+                if(!isBackup) AddToRecentFiles(path);
```

Comment:
Is there a reason that here the path is only being added to the recent files list, but at line 517 below in Save the model filename is being set and OnSaved is being called? It seems like they should be doing the same thing in both places maybe?

Reply:
They are actually a group of steps need to be done only after actual save, but forced to split since `AddToRecentFiles()` is a function lives in DynamoViewModel only, otherwise I would put them together